django/http/request.py

```diff
@@ -38,6 +40,62 @@ class RawPostDataException(Exception):
     pass
 
 
+class LimitChecker(object):
+    """
+    Stateful counter for enforcing a maximum number of parameters
+    """
+    def __init__(self, limit):
+        self.limit = limit
+        self.count = 0
+
+    def incr(self):
+        self.count += 1
+        if self.count > self.limit:
+            raise SuspiciousOperation('Too many fields')
+
+    def active(self):
+        return self.limit is not None
+
+    def start_split(self):
+        if self.count > 0:
+            self.count -= 1
+
+
+class SplitWrapper(str):
+    """
+    Exposes a custom split implementation when a max element
+    limit is configured that raises a SuspiciousOperation
+    exception when the limit is breached.
+    """
+
+    delims = ('&', ';')
+
+    def __new__(cls, query_string, limit_checker):
+        obj = str.__new__(cls, query_string)
+        obj.limit_checker = limit_checker
+        return obj
+
+    def __custom_split(self, sep):
+        sepsize = len(sep)
+        start = 0
+        result = []
+        self.limit_checker.start_split()
+        while True:
+            self.limit_checker.incr()
+            idx = self.__str__().find(sep, start)
+            if idx == -1:
+                result.append(SplitWrapper(self.__str__()[start:], self.limit_checker))
+                return result
+            result.append(SplitWrapper(self.__str__()[start:idx], self.limit_checker))
+            start = idx + sepsize
+
+    def split(self, sep=' ', maxsplit=-1):
+        if self.limit_checker.active() and sep in SplitWrapper.delims:
+            return self.__custom_split(sep)
+        else:
+            return super(SplitWrapper, self).split(sep, maxsplit)
+
```

Comment:
I'll tackle the inclusion of parse_qsl() next. Since there are two versions, one from python2 and another from python3, I may need to add both... :(